server/controllers/post-controller.ts

```diff
@@ -0,0 +1,103 @@
+import express from 'express';
+import Boom from '@hapi/boom';
+import { Request, Response, NextFunction, RequestHandler } from 'express-serve-static-core';
+import { Dependencies } from '../types';
+import { PostService } from '../service/post-service';
+import { UserService } from '../service/user-service';
+
+export class PostController {
+  notFound: Boom;
+  router: express.Router;
+  postService: PostService;
+  userService: UserService;
+  validatePost: RequestHandler;
+  validatePostPUT: RequestHandler;
+  validateParams: RequestHandler;
+  constructor(options: Dependencies) {
+    this.router = express.Router();
+    this.notFound = Boom.notFound("Post doesn't exist");
+    this.postService = options.postService;
+    this.userService = options.userService;
+    this.validatePost = options.validatePost;
+    this.validatePostPUT = options.validatePostPUT;
+    this.validateParams = options.validateParams;
+    this.initializeRoutes();
+  }
+
+  initializeRoutes() {
+    this.router.get('/', this.getAll.bind(this));
```

Comment:
I would use fat arrow functions to be able to get rid of binding. But you can choose which is better in your opinion 🤷‍♂